File: make_submission.R
make_submission <- function(name, dataframe) {
    ## Parameters
    ## ----------
    ##  name:               string, your name
    ##  submission_number:  int or string
    ##  dataframe:         data.frame [20000, 2], userId and 
    ##                          predicted probabilties on the test set
    
    # check input data frame appropriate shape
    if(nrow(dataframe) != 20000 | ncol(dataframe) != 2) {
        stop('data frame is wrong shape. Expecting [20000, 2]')
    # check input data for correct column names
    } else if(names(dataframe)[1] != 'userId') {
        stop('column 1 name is wrong. Expecting userId')
    }
    
    filename <- paste0(name, '.csv')
    write.csv(dataframe, filename, row.names = FALSE)
    
    return(paste(filename, 'created'))
}

File: make_submission.py
import pandas as pd
from time import time

def make_submission(name, data_frame):
    """
    Parameters:
    -----------
        name:               string, your name
        data_frame:         pandas DataFrame [20000, 2], userId and 
                                predicted probabilities on the test set
    """
    cnames = data_frame.columns.values
    if not isinstance(data_frame, pd.DataFrame):
        raise ValueError('Expecting a pandas DataFrame for argument 2')
    elif cnames[0] != 'userId':
        raise ValueError('Column name 0 incorrect, expecting userId, received ' + cnames[0])
        
    t0 = time()
    filename = name + '.csv'
    data_frame.to_csv(filename, index = False)
    return '{fname} written in {t:.{rd}f} ms'.format(fname = filename, t = (time() - t0) / 1000., rd = 3)